include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -948,6 +948,49 @@ namespace internal
 
   class MGTwoLevelTransferImplementation
   {
+    template <int dim, typename Number, typename MeshType>
+    static void
+    precompute_constraints(
+      const MeshType &                         dof_handler_coarse,
+      const dealii::AffineConstraints<Number> &constraint_coarse,
+      MGTwoLevelTransfer<dim, LinearAlgebra::distributed::Vector<Number>>
+        &transfer)
+    {
+      transfer.constraint_coarse.copy_from(constraint_coarse);
+
+      std::vector<types::global_dof_index> dependencies;
+
+      const auto locally_owned_dofs = dof_handler_coarse.locally_owned_dofs();
```

Comment:
clang-tidy wants to change this one (unnecessary copy):
```suggestion
      const auto &locally_owned_dofs = dof_handler_coarse.locally_owned_dofs();
```